synapse/cortex.py

```diff
@@ -881,7 +881,9 @@ async def initServiceStorage(self):
 
         self.maxnodes = self.conf.get('max:nodes')
         self.nodecount = 0
+
         self.migration = False
+        self.migration_lock = asyncio.Lock()
```

Comment:
Not really a perf issue, but why do we need this lock?

Reply:
Prevent the migration from running concurrently on startup of a leader.

From the Cell boot process:

3. initNexusSubsystem -> initCellActive ( on a leader ) -> creates schedcoro for _checkLayerModels THEN view / trigger tasks fire
4. initServiceRuntime -> blocking call of _checkLayerModels

There is nothing stopping the call in #4 from running concurrently with the schedcoro in #3. (edited)

Reply:
Do we even need the call to `_checkLayerModels()` in `initServiceRuntime` then? If the cell is active, `_checkLayerModels` will always be run from `initServiceActive`. We could remove the additional call to `_checkLayerModels` and the lock.

Reply:
If we want to retain the behavior which has the data model migrations fully executed on a regular Cortex startup we would need to keep the call in `initServiceRuntime`.